.travis.yml

```diff
@@ -1,5 +1,42 @@
 dist: trusty
+sudo: required
+
+matrix:
+    include:
+        - os: linux
+        - env: OS_TYPE=fedora
+        - env: INSTALL_REQUIREMENTS="dnf repolist; dnf install -y perl shellcheck devscripts sudo wget"
+
+        - os: linux
+        - env: OS_TYPE=opensuse
+        - env: INSTALL_REQUIREMENTS="zypper refresh; zypper in -y perl shellcheck devscripts sugo wget"
+
+        - os: linux
+        - env: OS_TYPE=dauchida/freebsd
+        - env: INSTALL_REQUIREMENTS="pkg install -y perl shellcheck devscripts sudo wget"
+
+        - os: linux
+        - env: OS_TYPE=debian:11
+        - env: INSTALL_REQUIREMENTS="sudo apt-get update -qq; sudo apt-get upgrade -y; sudo apt-get install -y -qq perl shellcheck devscripts sudo wget"
+
+        - os: linux
+        - env: OS_TYPE=ubuntu:22.04
+        - env: INSTALL_REQUIREMENTS="sudo apt-get update -qq; sudo apt-get upgrade -y; sudo apt-get install -y -qq perl shellcheck devscripts sudo wget"
+
+        - os: osx
+        - env: OS_TYPE=osx
+        - env: INSTALL_REQUIREMENTS="brew update; brew install perl shellcheck devscripts sudo wget"
+
 before_install:
-- sudo apt-get update -qq
-- sudo apt-get install -qq perl shellcheck devscripts
-script: prove
+    - [[ $TRAVIS_OS_NAME == "linux" ]] && docker pull ${OS_TYPE}
+
+install:
+    - [[ $TRAVIS_OS_NAME == "osx" ]] && docker run -d $OS /bin/sh -c "$INSTALL_REQUIREMENTS"
+    - [[ $TRAVIS_OS_NAME == "linux" ]] && docker run -d $OS /bin/sh -c "$INSTALL_REQUIREMENTS"
+    - docker run -d $OS /bin/sh -c "sudo wget https://github.com/EspaceNetworks/virtualmin-install/raw/master/virtualmin-install.sh"
```

Comment:
Should not hard code to your repository.
Are we able to use checked out file in source directory?

Reply:
Good point.  Have to look up the runtime variable in travis for "the URL to a file in the current checked out source we're running this test on" so that it'd work right from everywhere.